### PR TITLE
Adding slice notation for edges 

### PR DESCRIPTION
**Pull request by sidek**
This adds the functionality of accessing edges like node[1:2], as discussed in issue #158.

Only integer slices work -- I thought about making node['a':'b'] work, for named axes, but this seems like functionality more likely to create bugs than to be useful. I can update the PR to include this functionality if anyone thinks otherwise, however.

### CODE REVIEW

**Comment by sidek** [comment]
Oops! Will fix typing errors and report back. 

For type consistency elsewhere, it probably it isn't good to have get_edge possibly return a list, so this calls for some refactoring.

File: tensornetwork/network_components.py [hunk 1]
```diff
@@ -256,9 +257,20 @@ def has_nondangling_edge(self):
...
+  def __getitem__(self, key: slice) -> List["Edge"]:
+    return self.edges[key]
+  
+  @overload
   def __getitem__(self, key: Union[int, Text]) -> "Edge":
     return self.get_edge(key)
 
+  def __getitem__(self, key: Union[int, Text, slice]
+                 ) -> Union["Edge", List["Edge"]]:
+    if isinstance(key, slice):
```
Comment (Thenerdstation) [inline tensornetwork/network_components.py]: Isn't this redundant with the `@overload`?

Comment (sidek) [reply tensornetwork/network_components.py]: `@overload`-decorated function signatures are only read by the type checker, and must be followed by a non-overload decorated function signature, at least according to the typing documentation.

Really I wanted to write 
```
  @overload
  def __getitem__(self, key: slice) -> List["Edge"]:
    pass 

  @overload
  def __getitem__(self, key: Union[int, Text]) -> "Edge":
    pass
```
but pytype didn't like this, initially. 

It seems like pytype is fine with the above now that I've demanded travis use a more updated version of pytype, so I've merged that.

**Comment by Thenerdstation** [comment]
Not sure what's going on with the Google bot...

**Comment by sidek** [comment]
I signed it!

**Pull request merged**